<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Johannes Nicolaus Wibisana" />

<meta name="date" content="2019-12-26" />

<title>Gviz track viewer</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<script src="site_libs/elevate-section-attrs-2.0/elevate-section-attrs.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Tutorials by Nico</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    List of tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="alltuts.html">List of all tutorials</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">RNA-seq</li>
    <li class="dropdown-header">-</li>
    <li>
      <a href="tpm_calculation.html">TPM calculation</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Article Metadata</li>
  </ul>
</li>
<li>
  <a href="https://johannesnicolaus.github.io/index/">Personal website</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a>
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a>
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="aaa.com">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gviz track viewer</h1>
<h4 class="author">Johannes Nicolaus Wibisana</h4>
<h4 class="date">December 26, 2019</h4>

</div>


<div id="gviz-track-viewer-visualization-tutorial" class="section level2">
<h2>Gviz track viewer visualization tutorial</h2>
<p>This tutorial covers plotting of track file such as bigwig file for better visualization. This tutorial was constructed for The Laboratory of Cell Systems, Institute for Protein Research, Suita, Osaka, Japan.</p>
<p>The data used in this tutorial is ChIP-seq bigwig file obtained from</p>
<p>Load important libraries</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(Gviz)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">library</span>(rtracklayer)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="kw">library</span>(GenomicFeatures)</span></code></pre></div>
<p>Then we have to make txdb file.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>GTF_dir &lt;-<span class="st"> &quot;D:/Rmd_compilations/20191226_gviz/Mus_musculus.GRCm38.98.chr.gtf&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co"># mm10_txdb &lt;- makeTxDbFromGFF(GTF_dir,</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co">#                                 format=&quot;gtf&quot;,</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co">#                                 organism=&quot;Mus musculus&quot;,</span></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co">#                                 dbxrefTag = &quot;gene_name&quot;)</span></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="co"># </span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># </span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="co"># saveDb(mm10_txdb, &quot;mm10.txdb&quot;)</span></span>
<span id="cb2-10"><a href="#cb2-10"></a></span>
<span id="cb2-11"><a href="#cb2-11"></a>mm10_txdb &lt;-<span class="st"> </span><span class="kw">loadDb</span>(<span class="st">&quot;D:/Rmd_compilations/20191226_gviz/mm10.txdb&quot;</span>) </span></code></pre></div>
<p> </p>
<div id="plotting-genome" class="section level3">
<h3>Plotting genome</h3>
<p>Firstly, set the options as we are using Ensembl GTF file.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">options</span>(<span class="dt">ucscChromosomeNames =</span> F)</span></code></pre></div>
<p>Then we read the genome track and the track for the wanted gene region.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># read the genome, there is no need to put any argument</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>gtrack &lt;-<span class="st"> </span><span class="kw">GenomeAxisTrack</span>()</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># specify the start and end of chromosome (this has to be specified after looking at igv)</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>chr_no &lt;-<span class="st"> &quot;12&quot;</span> <span class="co"># chromosome number</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>chr_start &lt;-<span class="st"> </span><span class="dv">86330000</span> <span class="co"># start of region</span></span>
<span id="cb4-7"><a href="#cb4-7"></a>chr_end &lt;-<span class="st"> </span><span class="dv">86583978</span> <span class="co"># end of region</span></span>
<span id="cb4-8"><a href="#cb4-8"></a></span>
<span id="cb4-9"><a href="#cb4-9"></a></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co"># specify the start and end of the track you want to view with the chromosome number</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>gtTrack &lt;-<span class="st"> </span><span class="kw">GeneRegionTrack</span>(mm10_txdb,</span>
<span id="cb4-12"><a href="#cb4-12"></a>                           <span class="dt">chromosome=</span>chr_no, <span class="co"># chromosome number</span></span>
<span id="cb4-13"><a href="#cb4-13"></a>                           <span class="dt">start=</span>chr_start, <span class="co"># start of region</span></span>
<span id="cb4-14"><a href="#cb4-14"></a>                           <span class="dt">end=</span>chr_end, <span class="co"># end of region</span></span>
<span id="cb4-15"><a href="#cb4-15"></a>                           <span class="dt">transcriptAnnotation=</span><span class="st">&quot;gene_id&quot;</span>, <span class="co"># symbol is the gene symbol</span></span>
<span id="cb4-16"><a href="#cb4-16"></a>                           <span class="dt">fontsize.group=</span><span class="dv">20</span> <span class="co"># free to adjust font size</span></span>
<span id="cb4-17"><a href="#cb4-17"></a>                           )</span></code></pre></div>
<p>Try plotting the tracks</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">plotTracks</span>(gtTrack)</span></code></pre></div>
<p><img src="gviztutorial_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p> </p>
</div>
<div id="plotting-ideogram-track" class="section level3">
<h3>Plotting ideogram track</h3>
<p>Get the ideogram track. This can show the position of the track in the chromosome. This function automatically downloads annotation from database, therefore we need to specify the chromosome name in ucsc naming (with chr prefix). However, after getting the information, we need to return it back to without chr prefix naming.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a> itrack &lt;-<span class="st"> </span><span class="kw">IdeogramTrack</span>(<span class="dt">genome=</span><span class="st">&quot;mm10&quot;</span>, </span>
<span id="cb6-2"><a href="#cb6-2"></a>                       <span class="dt">chromosome=</span><span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>,chr_no), <span class="co"># specify chromosome in ucsc naming</span></span>
<span id="cb6-3"><a href="#cb6-3"></a>                       <span class="dt">from =</span>chr_start,</span>
<span id="cb6-4"><a href="#cb6-4"></a>                       <span class="dt">to=</span>chr_end)</span>
<span id="cb6-5"><a href="#cb6-5"></a> </span>
<span id="cb6-6"><a href="#cb6-6"></a>itrack<span class="op">@</span>chromosome &lt;-<span class="st"> </span>chr_no</span>
<span id="cb6-7"><a href="#cb6-7"></a></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co"># remove chr from chromosome naming</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="kw">levels</span>(itrack<span class="op">@</span>bandTable<span class="op">$</span>chrom) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;^chr&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">levels</span>(itrack<span class="op">@</span>bandTable<span class="op">$</span>chrom), <span class="dt">ignore.case=</span>T)</span></code></pre></div>
<p> </p>
</div>
<div id="importing-and-plotting-the-bigwig-files" class="section level3">
<h3>Importing and plotting the bigwig files</h3>
<p>Now we import the bigwig file of our wanted data</p>
<p>For ensembl annotation, the chromosome is named “1” instead of “chr1”, therefore there is a need to remove the chr string. However, the important thing is to have a consistent chromosome naming scheme (with or without chr for all of the annotation). In this tutorial, I will remove all the chr strings.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>bw_med1 &lt;-<span class="st"> </span><span class="kw">import.bw</span>(<span class="st">&quot;D:/Rmd_compilations/20191226_gviz/Med1.bigwig&quot;</span>, <span class="dt">as=</span><span class="st">&quot;GRanges&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2"></a>bw_h3k &lt;-<span class="st"> </span><span class="kw">import.bw</span>(<span class="st">&quot;D:/Rmd_compilations/20191226_gviz/H3K27Ac.bigwig&quot;</span>, <span class="dt">as=</span><span class="st">&quot;GRanges&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co"># change chr name to without chr (this depends on the data)</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>bw_med1<span class="op">@</span>seqnames<span class="op">@</span>values &lt;-<span class="st"> </span><span class="kw">str_replace_all</span>(bw_med1<span class="op">@</span>seqnames<span class="op">@</span>values, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.factor</span>()</span>
<span id="cb7-6"><a href="#cb7-6"></a>bw_h3k<span class="op">@</span>seqnames<span class="op">@</span>values &lt;-<span class="st"> </span><span class="kw">str_replace_all</span>(bw_h3k<span class="op">@</span>seqnames<span class="op">@</span>values, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.factor</span>()</span>
<span id="cb7-7"><a href="#cb7-7"></a></span>
<span id="cb7-8"><a href="#cb7-8"></a>bw_med1<span class="op">@</span>seqinfo<span class="op">@</span>seqnames &lt;-<span class="st"> </span><span class="kw">str_replace_all</span>(bw_med1<span class="op">@</span>seqinfo<span class="op">@</span>seqnames, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-9"><a href="#cb7-9"></a>bw_h3k<span class="op">@</span>seqinfo<span class="op">@</span>seqnames &lt;-<span class="st"> </span><span class="kw">str_replace_all</span>(bw_h3k<span class="op">@</span>seqinfo<span class="op">@</span>seqnames, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;&quot;</span>) </span></code></pre></div>
<p>Then we specify which part of data we want to show.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># track 1 med1</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>med1_track &lt;-<span class="st"> </span><span class="kw">DataTrack</span>(<span class="dt">range=</span>bw_med1,</span>
<span id="cb8-3"><a href="#cb8-3"></a>                     <span class="dt">chromosome=</span>chr_no,</span>
<span id="cb8-4"><a href="#cb8-4"></a>                     <span class="dt">from =</span>chr_start,</span>
<span id="cb8-5"><a href="#cb8-5"></a>                     <span class="dt">to=</span>chr_end,</span>
<span id="cb8-6"><a href="#cb8-6"></a>                     <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">290</span>),</span>
<span id="cb8-7"><a href="#cb8-7"></a>                     <span class="dt">col.histogram=</span><span class="kw">c</span>(<span class="st">&quot;#FDE725FF&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8"></a>)</span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co"># track 2 h3k</span></span>
<span id="cb8-10"><a href="#cb8-10"></a>h3k_track &lt;-<span class="st"> </span><span class="kw">DataTrack</span>(<span class="dt">range=</span>bw_h3k,</span>
<span id="cb8-11"><a href="#cb8-11"></a>                         <span class="dt">chromosome=</span>chr_no,</span>
<span id="cb8-12"><a href="#cb8-12"></a>                         <span class="dt">from =</span>chr_start,</span>
<span id="cb8-13"><a href="#cb8-13"></a>                         <span class="dt">to=</span>chr_end,</span>
<span id="cb8-14"><a href="#cb8-14"></a>                         <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">240</span>),</span>
<span id="cb8-15"><a href="#cb8-15"></a>                         <span class="dt">col.histogram=</span><span class="kw">c</span>(<span class="st">&quot;#440154FF&quot;</span>)</span>
<span id="cb8-16"><a href="#cb8-16"></a>)</span></code></pre></div>
<p>Before plotting the track, we want to convert the gene id to gene symbol.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>convertensembl &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">x =</span> gtTrack){</span>
<span id="cb9-2"><a href="#cb9-2"></a>  <span class="kw">require</span>(biomaRt)</span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="kw">require</span>(org.Mm.eg.db)</span>
<span id="cb9-4"><a href="#cb9-4"></a>  </span>
<span id="cb9-5"><a href="#cb9-5"></a>  mouse &lt;-<span class="st"> </span><span class="kw">useMart</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="dt">dataset =</span> <span class="st">&quot;mmusculus_gene_ensembl&quot;</span>)</span>
<span id="cb9-6"><a href="#cb9-6"></a>  </span>
<span id="cb9-7"><a href="#cb9-7"></a>  convertedgene &lt;-<span class="st"> </span><span class="kw">getBM</span>(<span class="dt">attributes =</span> <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;external_gene_name&quot;</span>),</span>
<span id="cb9-8"><a href="#cb9-8"></a>                         <span class="dt">filters =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb9-9"><a href="#cb9-9"></a>                         <span class="dt">values =</span> x<span class="op">@</span>range<span class="op">@</span>elementMetadata<span class="op">@</span>listData<span class="op">$</span>gene,</span>
<span id="cb9-10"><a href="#cb9-10"></a>                         <span class="dt">mart =</span> mouse)</span>
<span id="cb9-11"><a href="#cb9-11"></a></span>
<span id="cb9-12"><a href="#cb9-12"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(convertedgene)){</span>
<span id="cb9-13"><a href="#cb9-13"></a>    x<span class="op">@</span>range<span class="op">@</span>elementMetadata<span class="op">@</span>listData<span class="op">$</span>gene &lt;-<span class="st"> </span>x<span class="op">@</span>range<span class="op">@</span>elementMetadata<span class="op">@</span>listData<span class="op">$</span>gene <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb9-14"><a href="#cb9-14"></a><span class="st">      </span><span class="kw">str_replace_all</span>(convertedgene[i,<span class="dv">1</span>], convertedgene[i,<span class="dv">2</span>]) </span>
<span id="cb9-15"><a href="#cb9-15"></a>    </span>
<span id="cb9-16"><a href="#cb9-16"></a>  }</span>
<span id="cb9-17"><a href="#cb9-17"></a>  <span class="kw">return</span>(x)</span>
<span id="cb9-18"><a href="#cb9-18"></a>}</span>
<span id="cb9-19"><a href="#cb9-19"></a></span>
<span id="cb9-20"><a href="#cb9-20"></a></span>
<span id="cb9-21"><a href="#cb9-21"></a></span>
<span id="cb9-22"><a href="#cb9-22"></a><span class="co"># convert ensembl id to gene name</span></span>
<span id="cb9-23"><a href="#cb9-23"></a>gtTrack &lt;-<span class="st"> </span><span class="kw">convertensembl</span>(gtTrack)</span></code></pre></div>
</div>
</div>
<div id="plot-all-tracks-in-one-plot" class="section level2">
<h2>Plot all tracks in one plot</h2>
<p>Finally combine the tracks</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>combinetracks &lt;-<span class="st"> </span><span class="kw">plotTracks</span>(<span class="kw">c</span>(gtrack, itrack, med1_track, h3k_track, gtTrack),</span>
<span id="cb10-2"><a href="#cb10-2"></a>           <span class="dt">transcriptAnnotation=</span><span class="st">&quot;gene&quot;</span>,</span>
<span id="cb10-3"><a href="#cb10-3"></a>           <span class="dt">type=</span><span class="st">&quot;hist&quot;</span>,</span>
<span id="cb10-4"><a href="#cb10-4"></a>           <span class="dt">from =</span>chr_start,</span>
<span id="cb10-5"><a href="#cb10-5"></a>           <span class="dt">to=</span>chr_end,</span>
<span id="cb10-6"><a href="#cb10-6"></a>           <span class="dt">background.title =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb10-7"><a href="#cb10-7"></a>           <span class="dt">fontcolor =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb10-8"><a href="#cb10-8"></a>           <span class="dt">col.axis=</span><span class="st">&quot;black&quot;</span>,</span>
<span id="cb10-9"><a href="#cb10-9"></a>           <span class="dt">fontsize=</span><span class="dv">15</span>,</span>
<span id="cb10-10"><a href="#cb10-10"></a>           <span class="dt">showTitle=</span>F,</span>
<span id="cb10-11"><a href="#cb10-11"></a>           <span class="dt">margin=</span><span class="dv">40</span>,</span>
<span id="cb10-12"><a href="#cb10-12"></a>           <span class="dt">innerMargin =</span> <span class="dv">10</span></span>
<span id="cb10-13"><a href="#cb10-13"></a>          </span>
<span id="cb10-14"><a href="#cb10-14"></a>)</span></code></pre></div>
<p><img src="gviztutorial_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="johannesnicolaus.github.io/index.html">Johannes Nicolaus Wibisana</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>johannes.nicolaus@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://twitter.com/otjin" class="fa fa-twitter"></a>
    <a href="https://www.linkedin.com/in/johannes-nicolaus-wibisana-95a375168/" class="fa fa-linkedin"></a>
    <a href="https://github.com/johannesnicolaus/" class="fa fa-github"></a>
</p>

&nbsp;


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
