<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Johannes Nicolaus Wibisana" />


<title>TPM calculation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<script src="site_libs/elevate-section-attrs-2.0/elevate-section-attrs.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Tutorials by Nico</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    List of tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="alltuts.html">List of all tutorials</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">RNA-seq</li>
    <li class="dropdown-header">-</li>
    <li>
      <a href="tpm_calculation.html">TPM calculation</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="gviztutorial.html">Gviz tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://johannesnicolaus.github.io/index/">Personal website</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/johannesnicolaus/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/otjin">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/johannes-nicolaus-wibisana-95a375168/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TPM calculation</h1>
<h4 class="author">Johannes Nicolaus Wibisana</h4>
<h4 class="date">11/4/2019</h4>

</div>


<div id="tpm-transcripts-per-million-calculation" class="section level2">
<h2>TPM (transcripts per million) calculation</h2>
<p>This tutorial covers transcripts per million calculation with library size normalization using edgeR. This tutorial was constructed for The Laboratory of Cell Systems, Institute for Protein Research, Suita, Osaka, Japan.</p>
<p>prep our library</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">library</span>(edgeR)</span></code></pre></div>
<pre><code>## Warning: package &#39;edgeR&#39; was built under R version 3.6.1</code></pre>
<p>let’s read our raw counts data</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>counts &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="dt">file =</span> <span class="st">&quot;D:/lab_rmd/tpm/counts_trimmed.csv&quot;</span>)</span></code></pre></div>
<p>Here is how the data should look like</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">head</span>(counts) </span></code></pre></div>
<pre><code>## # A tibble: 6 x 10
##   Geneid          Chr               Start                                      End                                       Strand            Length B00_A01 B00_A02 B00_A03 B00_A04
##   &lt;chr&gt;           &lt;chr&gt;             &lt;chr&gt;                                      &lt;chr&gt;                                     &lt;chr&gt;              &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 ENSGALG0000005… 1;1;1;1;1;1;1;1;… 5273;5273;5536;6755;6755;6755;8674;8674;8… 5524;5524;5952;6829;6829;6829;8775;8904;… -;-;-;-;-;-;-;-;…   2018       0       0       0       0
## 2 ENSGALG0000005… 1;1;1;1;1         9441;9681;9768;10182;10256                 10053;9683;10070;14305;14305              +;+;+;+;+           4754       0       0       0       0
## 3 ENSGALG0000004… 1;1;1;1;1         27209;32230;32540;33081;33287              27503;32331;32721;33187;33555             +;+;+;+;+            955     105      73      35      42
## 4 ENSGALG0000005… 1;1;1;1;1         31439;32230;53984;54190;242817             31484;32331;54090;54263;243537            +;+;+;+;+           1050     285     220     124     230
## 5 ENSGALG0000004… 1;1;1;1;1;1;1;1;1 39057;39080;39967;39967;40433;40433;40823… 39867;39867;40073;40073;40924;40614;4092… -;-;-;-;-;-;-;-;-   2161     224     138     134     178
## 6 ENSGALG0000004… 1                 58427                                      58617                                     +                    191       0       0       0       0</code></pre>
<p>Remove Chr, Start, End, Strand. Only keep geneid, length and samples</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>counts &lt;-<span class="st"> </span>counts[,<span class="op">!</span><span class="kw">colnames</span>(counts) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Chr&quot;</span>, <span class="st">&quot;Start&quot;</span>, <span class="st">&quot;End&quot;</span>, <span class="st">&quot;Strand&quot;</span>)]</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">head</span>(counts)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 6
##   Geneid             Length B00_A01 B00_A02 B00_A03 B00_A04
##   &lt;chr&gt;               &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 ENSGALG00000054818   2018       0       0       0       0
## 2 ENSGALG00000053455   4754       0       0       0       0
## 3 ENSGALG00000045540    955     105      73      35      42
## 4 ENSGALG00000051297   1050     285     220     124     230
## 5 ENSGALG00000042023   2161     224     138     134     178
## 6 ENSGALG00000047594    191       0       0       0       0</code></pre>
<p>Divide the sample read count by the length of the gene multiplied by 1000</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>counts_tpm &lt;-<span class="st"> </span>(counts[,<span class="dv">3</span><span class="op">:</span><span class="kw">ncol</span>(counts)]<span class="op">/</span>counts<span class="op">$</span>Length) <span class="op">*</span><span class="dv">1000</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="co">#return to tible just because I like tible</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>counts_tpm &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="dt">Geneid =</span> counts<span class="op">$</span>Geneid, counts_tpm) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()</span></code></pre></div>
<p>Get the normalization factor (in this case RLE)</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>normfactor &lt;-<span class="st"> </span><span class="kw">DGEList</span>(<span class="dt">counts =</span> counts_tpm[,<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(counts_tpm)],</span>
<span id="cb9-2"><a href="#cb9-2"></a>                        <span class="dt">group =</span> <span class="kw">colnames</span>(counts_tpm[,<span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(counts_tpm)])</span>
<span id="cb9-3"><a href="#cb9-3"></a>                        )</span>
<span id="cb9-4"><a href="#cb9-4"></a></span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="co">#you can change the normalization method here</span></span>
<span id="cb9-6"><a href="#cb9-6"></a>normfactor &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(normfactor, <span class="dt">method=</span><span class="st">&quot;RLE&quot;</span>)</span>
<span id="cb9-7"><a href="#cb9-7"></a>  </span>
<span id="cb9-8"><a href="#cb9-8"></a>normfactor_samples &lt;-<span class="st"> </span>normfactor<span class="op">$</span>samples</span>
<span id="cb9-9"><a href="#cb9-9"></a>  </span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#multiply normalization factor with the library size</span></span>
<span id="cb9-11"><a href="#cb9-11"></a>normfactor_samples<span class="op">$</span>normlib &lt;-<span class="st"> </span>normfactor_samples<span class="op">$</span>lib.size<span class="op">*</span>normfactor_samples<span class="op">$</span>norm.factors</span></code></pre></div>
<p>Loop to get the final TPM counts table</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">dim</span>(counts_tpm)[<span class="dv">2</span>]<span class="op">-</span><span class="dv">1</span>)){</span>
<span id="cb10-2"><a href="#cb10-2"></a>    </span>
<span id="cb10-3"><a href="#cb10-3"></a>  counts_tpm[,i<span class="op">+</span><span class="dv">1</span>] &lt;-<span class="st"> </span>(counts_tpm[,i<span class="op">+</span><span class="dv">1</span>]<span class="op">/</span>normfactor_samples<span class="op">$</span>normlib[i])<span class="op">*</span><span class="dv">1000000</span></span>
<span id="cb10-4"><a href="#cb10-4"></a>    </span>
<span id="cb10-5"><a href="#cb10-5"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">head</span>(counts_tpm)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 5
##   Geneid             B00_A01 B00_A02 B00_A03 B00_A04
##   &lt;fct&gt;                &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 ENSGALG00000054818     0       0       0       0  
## 2 ENSGALG00000053455     0       0       0       0  
## 3 ENSGALG00000045540    54.6    49.4    37.3    33.1
## 4 ENSGALG00000051297   135.    135.    120.    165. 
## 5 ENSGALG00000042023    51.5    41.3    63.1    61.9
## 6 ENSGALG00000047594     0       0       0       0</code></pre>
<!-- ## Comparison between normalized and unnormalized -->
<!-- You can also embed plots, for example: -->
<!-- ```{r pressure, echo=FALSE} -->
<!-- plot(pressure) -->
<!-- ``` -->
<!-- Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot. -->
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by Johannes Nicolaus Wibisana</p>
<p style="text-align: center;"><span style="color: #808080;"><em>johannes.nicolaus@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://twitter.com/otjin" class="fa fa-twitter"></a>
    <a href="https://www.linkedin.com/in/johannes-nicolaus-wibisana-95a375168/" class="fa fa-linkedin"></a>
    <a href="https://github.com/johannesnicolaus/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
